MinecraftClient/Inventory/Item.cs

```diff
@@ -70,5 +70,24 @@ public string DisplayName
                 return null;
             }
         }
+
+        /// <summary>
+        /// Check item is a food
+        /// </summary>
+        /// <returns>True if is a food</returns>
+        public bool IsFood()
+        {
+            // non-poison and stackable food
+            // remarks: auto eat may works with non-stackable food <- not tested
+            int[] foods = { 524, 765, 821, 823, 562, 763, 680, 629, 801, 585, 788, 630, 670, 674, 588, 587, 768, 673, 764, 777, 677, 679, 625, 800, 584, 787, 626, 678, 876, 627 };
+            if (foods.Contains((int)Type))
```

Comment:
Just use `if (Type.IsFood())` after defining the extension method (see below). Then you can remove the `foods` array completely.

Reply:
Seems like I totally forgot about the array here :P